catalog/mgmp/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/mgmp/MgmpTransformer.java

```diff
@@ -0,0 +1,681 @@
+                MgmpConstants.NIIRS_PATH,
+                MgmpConstants.NIIRS_RATING_PATH,
+                MgmpConstants.RESOURCE_SECURITY_PATH,
+                MgmpConstants.METADATA_SECURITY_PATH,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_CRS_WKT_TYPE_PATH)
+            .map(this::toPath)
+            .collect(Collectors.toCollection(LinkedHashSet::new));
+    paths.addAll(super.buildPaths());
+    return paths;
+  }
+
+  @Override
+  public MetacardImpl toMetacard(final XstreamPathValueTracker pathValueTracker, String id) {
+    MetacardImpl metacard = new MetacardImpl(mgmpMetacardType);
+    super.setMetacardAttributes(metacard, id, pathValueTracker);
+    addMetacardIsrInformation(pathValueTracker, metacard);
+    addMetacardNiirsInformation(pathValueTracker, metacard);
+    addMetacardLanguage(pathValueTracker, metacard);
+    addMetacardCloudCoverPercentage(pathValueTracker, metacard);
+    addMetacardResourceSecurityClassification(pathValueTracker, metacard);
+    addMetacardResourceSecurityReleasability(pathValueTracker, metacard);
+    addMetacardResourceSecurityOriginatorClassification(pathValueTracker, metacard);
+    addMetacardMetadataSecurityReleasability(pathValueTracker, metacard);
+    addMetacardMetadataSecurityOriginatorClassification(pathValueTracker, metacard);
+    addMetacardMetadataSecurityClassification(pathValueTracker, metacard);
+    addMetacardDataQuality(metacard);
+    addMetacardPublishedDate(metacard);
+    return metacard;
+  }
+
+  @Override
+  public void addMetacardFormat(
+      final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+    String format = pathValueTracker.getFirstValue(toPath(MgmpConstants.FORMAT_PATH));
+    String formatVersion = pathValueTracker.getFirstValue(toPath(GmdConstants.FORMAT_VERSION_PATH));
+    if (StringUtils.isNotEmpty(format)) {
+      metacard.setAttribute(Media.FORMAT, format);
+      if (StringUtils.isNotEmpty(formatVersion)) {
+        metacard.setAttribute(Media.FORMAT_VERSION, formatVersion);
+      }
+    }
+  }
+
+  @Override
+  public void addMetacardLanguage(
+      final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+    List<String> languageList = pathValueTracker.getAllValues(toPath(MgmpConstants.LANGUAGE_PATH));
+    if (CollectionUtils.isNotEmpty(languageList)) {
+      languageList = filterResourceLanguages(languageList);
+      metacard.setAttribute(Core.LANGUAGE, (Serializable) languageList);
+    } else {
+      metacard.setAttribute(Core.LANGUAGE, (Serializable) MgmpConstants.DEFAULT_LANGUAGE);
+    }
+  }
+
+  @Override
+  public void addMetacardCrs(
+      final XstreamPathValueTracker pathValueTracker, MetacardImpl metacard) {
+
+    super.addMetacardCrs(pathValueTracker, metacard);
+
+    ArrayList<String> mgmpTypes = new ArrayList<>();
+    addValuesFromPathToListIfPresent(
+        pathValueTracker,
+        mgmpTypes,
+        MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_NUMBER_PATH,
```

Comment:
✏️ I think if you extracted the constant groups here and in some of the other methods, you could clean up a lot of these method calls. It looks like all the other parameters are the same.